app/src/main/java/com/duckduckgo/app/location/ui/SiteLocationPermissionDialog.kt

```diff
@@ -0,0 +1,176 @@
+
+        val rootView = layoutInflater.inflate(R.layout.content_site_location_permission_dialog, null)
+
+        val title = rootView.find<TextView>(R.id.sitePermissionDialogTitle)
+        val favicon = rootView.find<ImageView>(R.id.sitePermissionDialogFavicon)
+        val allowAlways = rootView.find<TextView>(R.id.siteAllowAlwaysLocationPermission)
+        val allowOnce = rootView.find<TextView>(R.id.siteAllowOnceLocationPermission)
+        val denyOnce = rootView.find<TextView>(R.id.siteDenyOnceLocationPermission)
+        val denyAlways = rootView.find<TextView>(R.id.siteDenyAlwaysLocationPermission)
+        val extraDivider = rootView.find<View>(R.id.siteAllowOnceLocationPermissionDivider)
+        val anotherDivider = rootView.find<View>(R.id.siteDenyLocationPermissionDivider)
+
+        val alertDialog = AlertDialog.Builder(requireActivity(), R.style.AlertDialogTheme)
+            .setView(rootView)
+
+        validateBundleArguments()
+        populateTitle(title)
+        populateFavicon(favicon)
+        configureListeners(allowAlways, allowOnce, denyOnce, denyAlways)
+        hideExtraViews(allowOnce, denyOnce, extraDivider, anotherDivider)
+        makeCancellable()
+
+        return alertDialog.create()
+    }
+
+    private fun populateTitle(title: TextView) {
+        arguments?.let { args ->
+            val originUrl = args.getString(KEY_REQUEST_ORIGIN)
+            val dialogTitle = getString(R.string.preciseLocationSiteDialogTitle, originUrl.website())
+            title.text = dialogTitle
+        }
+    }
+
+    private fun populateFavicon(imageView: ImageView) {
+        arguments?.let { args ->
+            val originUrl = args.getString(KEY_REQUEST_ORIGIN)
+            val faviconUrl = Uri.parse(originUrl).faviconLocation()
+
+            GlideApp.with(requireContext())
+                .load(faviconUrl)
+                .error(R.drawable.ic_globe_gray_16dp)
+                .into(imageView)
+        }
+    }
+
+    private fun configureListeners(
+        allowAlways: TextView,
+        allowOnce: TextView,
+        denyOnce: TextView,
+        denyAlways: TextView
+    ) {
+        arguments?.let { args ->
+            val originUrl = args.getString(KEY_REQUEST_ORIGIN)
+            allowAlways.setOnClickListener {
+                dismiss()
+                listener.onSiteLocationPermissionSelected(originUrl, LocationPermissionType.ALLOW_ALWAYS)
+            }
+            allowOnce.setOnClickListener {
+                dismiss()
+                listener.onSiteLocationPermissionSelected(originUrl, LocationPermissionType.ALLOW_ONCE)
+            }
+            denyOnce.setOnClickListener {
+                dismiss()
+                listener.onSiteLocationPermissionSelected(originUrl, LocationPermissionType.DENY_ONCE)
+            }
+            denyAlways.setOnClickListener {
+                dismiss()
+                listener.onSiteLocationPermissionSelected(originUrl, LocationPermissionType.DENY_ALWAYS)
```

Comment:
lint warnings on nullability of `originUrl`. If we're confident it'll never be null, we should make that more explicit. With the warnings, it isn't clear if we're sure or just missed the warning

Reply:
yeah, as described above we are safe to make it explicit.